telegram/src/main/kotlin/com/github/kotlintelegrambot/network/serialization/adapter/TelegramFileAdapter.kt

```diff
@@ -0,0 +1,18 @@
+package com.github.kotlintelegrambot.network.serialization.adapter
+
+import com.github.kotlintelegrambot.entities.TelegramFile
+import com.github.kotlintelegrambot.entities.TelegramFile.ByFile
+import com.github.kotlintelegrambot.entities.TelegramFile.ByFileId
+import com.github.kotlintelegrambot.entities.TelegramFile.ByUrl
+import com.google.gson.JsonElement
+import com.google.gson.JsonSerializationContext
+import com.google.gson.JsonSerializer
+import java.lang.reflect.Type
+
+class TelegramFileAdapter : JsonSerializer<TelegramFile> {
+    override fun serialize(src: TelegramFile, typeOfSrc: Type, context: JsonSerializationContext): JsonElement = when (src) {
+        is ByFileId -> context.serialize(src.fileId, String::class.java)
+        is ByUrl -> context.serialize(src.url, String::class.java)
+        is ByFile -> context.serialize("attach://${src.file.name}")
+    }
```

Comment:
This is the way the different types of file must be serialized for the `sendMediaGroup` operation

`attach://filename` indicates that a file with that name is being sent as part of the multipart request body (it's the approach Telegram has stablished for this case)